File: BE/src/main/java/com/za/testexe/service/SpeseService.java
package com.za.testexe.service;

import com.za.testexe.model.dto.request.spese.SpeseRequest;
import com.za.testexe.model.dto.response.spese.SpeseResponse;

import java.util.List;

public interface SpeseService {
    SpeseResponse saveSpesa(SpeseRequest request);
    List<SpeseResponse> getSpese();
    void deleteSpesa(Integer idSpesa);
}

File: FE/src/app/services/spese.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { ApiResponse } from '../models/utlils.model';
import { SpeseModel } from '../models/spese.model';
import { API_BASE } from '../core/api.config';

@Injectable({
  providedIn: 'root',
})
export class SpeseService {
  private http = inject(HttpClient);
  private baseUrl = `${API_BASE}/api/spese`;

  saveSpesa(request: SpeseModel): Observable<ApiResponse<SpeseModel>> {
    return this.http.post<ApiResponse<SpeseModel>>(`${this.baseUrl}/salva`, request);
  }

  getSpese(): Observable<ApiResponse<SpeseModel[]>> {
    return this.http.get<ApiResponse<SpeseModel[]>>(`${this.baseUrl}/lista`);
  }

  deleteSpesa(idSpesa: number): Observable<ApiResponse<void>> {
    return this.http.delete<ApiResponse<void>>(`${this.baseUrl}/elimina/${idSpesa}`);
  }
}
